src/translib/acl_app.go

```diff
@@ -520,52 +584,55 @@ func (app *AclApp) processCommonToplevelPath(d *db.DB, acl *ocbinds.OpenconfigAc
 			return err
 		}
 		err = d.DeleteTable(app.aclTs)
-	case GET:
-		ygot.BuildEmptyTree(acl)
-		err = app.convertDBAclToInternal(d, db.Key{})
-		if err != nil {
-			return err
-		}
-		app.convertInternalToOCAcl("", acl.AclSets, nil)
-		if isTopmostPath {
-			err = app.getAllBindingsInfo(d)
-		}
 	}
 	return err
 }
 
 /***********    These are Translation Helper Function   ***********/
-func (app *AclApp) convertDBAclRulesToInternal(dbCl *db.DB, aclName string, seqId int64, ruleKey db.Key) error {
+func (app *AclApp) convertDBAclRulesToInternal(dbs [db.MaxDB]*db.DB, aclName string, seqId int64, ruleKey db.Key) error {
+	dbCl := dbs[db.ConfigDB]
+	dbCo := dbs[db.CountersDB]
+
+
 	var err error
 	if seqId != -1 {
 		ruleKey.Comp = []string{aclName, "RULE_" + strconv.FormatInt(int64(seqId), 10)}
 	}
 	if ruleKey.Len() > 1 {
 		ruleName := ruleKey.Get(1)
-        ruleData, err := dbCl.GetEntry(app.ruleTs, ruleKey)
-        if err != nil {
-            return err
-        }
-        if app.ruleTableMap[aclName] == nil {
-            app.ruleTableMap[aclName] = make(map[string]db.Value)
-        }
-        app.ruleTableMap[aclName][ruleName] = ruleData
+		ruleData, err := dbCl.GetEntry(app.ruleTs, ruleKey)
+		if err != nil {
+			return err
+		}
+		if app.ruleTableMap[aclName] == nil {
+			app.ruleTableMap[aclName] = make(map[string]db.Value)
+		}
+		counterData, err := dbCo.GetEntry(app.counterTs, ruleKey)
+		if err == nil {
+			for k, x := range counterData.Field {
+				ruleData.Field[k] = x
+				// log.Info("Packets count " ,x,  " found in  COUNTER db for rulekey ", ruleKey.Comp) 
+			}
+		}
+		app.ruleTableMap[aclName][ruleName] = ruleData 
 	} else {
 		ruleKeys, err := dbCl.GetKeys(app.ruleTs)
 		if err != nil {
 			return err
 		}
 		for i, _ := range ruleKeys {
 			if aclName == ruleKeys[i].Get(0) {
-				app.convertDBAclRulesToInternal(dbCl, aclName, -1, ruleKeys[i])
+				app.convertDBAclRulesToInternal(dbs, aclName, -1, ruleKeys[i])
```

Comment:
Could you please handle the error returned?

Reply:
This is a recursive function  and we want to continue for other entries. I will print a  log message where the main handling happens for debug-ability.